v2/tools/generator/internal/armconversion/convert_to_arm_function_builder.go

```diff
@@ -100,7 +100,9 @@ func (builder *convertToARMBuilder) functionBodyStatements() []dst.Stmt {
 	result = append(
 		result,
 		astbuilder.ReturnIfNil(dst.NewIdent(builder.receiverIdent), astbuilder.Nil(), astbuilder.Nil()))
-	result = append(result, astbuilder.NewVariable(builder.resultIdent, builder.armTypeIdent))
+
+	result = append(result, astbuilder.ShortDeclaration(builder.resultIdent, astbuilder.AddrOf(astbuilder.NewCompositeLiteralBuilder(dst.NewIdent(builder.armTypeIdent)).Build())))
```

Comment:
This coule be simplified by using `astbuilder.Statements()` to build the final `[]dst.Stmt`.

Reply:
Agreed that might help some, but have kept this as is for now as it matches the pattern of the rest of the file (which was written before `astbuilder.Statements` existed). I think it's probably more important to match the local style of the file, especially in this code which is pretty complex. The alternative would be to fix the whole file, but that'd be a change for another PR